### ISSUE TITLE
GTimer 'timerid' parameter validation bug

### ISSUE THREAD

**Issue by xavier862** [ambiot/ambd_arduino#61]
Here is how [GTimer.cpp](https://github.com/ambiot/ambd_arduino/blob/fc5763bdb97156c4a2e211df9f392c10574c5588/Arduino_package/hardware/libraries/GTimer/GTimer.cpp) functions currently validate the `timerid` parameter:
```
    if (timerid > GTIMER_MAX) {
        return;
    }
```
As `timerid` is used to index the `timer_mapping[GTIMER_MAX]`array, obviously the case `timerid == GTIMER_MAX` should be rejected. Hence the validation test should be
```
    if (timerid >= GTIMER_MAX) {
        return;
    }
```

Regards

**Comment by xidameng** [comment]
> Here is how [GTimer.cpp](https://github.com/ambiot/ambd_arduino/blob/fc5763bdb97156c4a2e211df9f392c10574c5588/Arduino_package/hardware/libraries/GTimer/GTimer.cpp) functions currently validate the `timerid` parameter:
> 
> ```
>     if (timerid > GTIMER_MAX) {
>         return;
>     }
> ```
> 
> As `timerid` is used to index the `timer_mapping[GTIMER_MAX]`array, obviously the case `timerid == GTIMER_MAX` should be rejected. Hence the validation test should be
> 
> ```
>     if (timerid >= GTIMER_MAX) {
>         return;
>     }
> ```
> 
> Regards

You are right! if not the ID 4 will pass the test which is not godd

**Comment by xidameng** [comment]
This issue is already fixed in [51ad75b](https://github.com/ambiot/ambd_arduino/commit/51ad75bb17315d0f6dcb3bb9949aa3727317c86b), thanks for your contribution!
